govcd/orgvdcnetwork.go

```diff
@@ -193,3 +193,60 @@ func (vdc *Vdc) CreateOrgVDCNetwork(networkConfig *types.OrgVDCNetwork) (Task, e
 	}
 	return Task{}, fmt.Errorf("network creation failed: no operational link found")
 }
+
+// FindEdgeGatewayNameByNetwork searches the VDC for a connection between an edge gateway and a given network.
+// On success, returns the name of the edge gateway
+//
+// CAVEAT: this function should only be called on routed networks.
+// There is a slim chance that this function returns a misleading result when BOTH the following
+// conditions are met:
+// (a) we query an isolated or direct network instead of a routed one
+// AND
+// (b) the network has the same name of an existing edge gateway
+// In normal operations (like retrieving the edge gateway name for a routed network data source) this will work
+// as expected.
+// If this function is used to detect the network type (i.e. if it is connected to an edge gateway it's routed)
+// it may fail its purpose if the above mentioned circumstance apply.
+func (vdc *Vdc) FindEdgeGatewayNameByNetwork(networkName string) (string, error) {
```

Comment:
If comment are right, then function name should be named `FindEdgeGatewayNameByRoutedNetwork`

Reply:
I think the name is long enough as it is.
It was @lvirbalas suggestion.

Reply:
@lvirbalas This suggestions implies that I can use any network.

Reply:
As only routed has EG, I'm OK with leaving it as "short" as it is right now :)